### PR TITLE
[MINOR][DOCS] Minor doc fixes related with doc build and uses script dir in SQL doc gen script

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes both:

- Add information about Javadoc, SQL docs and few more information in `docs/README.md` and a comment in `docs/_plugins/copy_api_dirs.rb` related with Javadoc.

- Adds some commands so that the script always runs the SQL docs build under `./sql` directory (for directly running `./sql/create-docs.sh` in the root directory).

## How was this patch tested?

Manual tests with `jekyll build` and `./sql/create-docs.sh` in the root directory.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -9,19 +9,22 @@ documentation yourself. Why build it yourself? So that you have the docs that co
...
-    $ sudo gem install jekyll jekyll-redirect-from pygments.rb
-    $ sudo pip install Pygments
-    # Following is needed only for generating API docs
-    $ sudo pip install sphinx pypandoc mkdocs
-    $ sudo Rscript -e 'install.packages(c("knitr", "devtools", "roxygen2", "testthat", "rmarkdown"), repos="http://cran.stat.ucla.edu/")'
+$ sudo gem install jekyll jekyll-redirect-from pygments.rb
+$ sudo pip install Pygments
+# Following is needed only for generating API docs
+$ sudo pip install sphinx pypandoc mkdocs
+$ sudo Rscript -e 'install.packages(c("knitr", "devtools", "roxygen2", "testthat", "rmarkdown"), repos="http://cran.stat.ucla.edu/")'
```
Comment (HyukjinKwon) [inline docs/README.md]: I think we mistakenly make this indented. For example, these were all intended identically - https://github.com/apache/spark/blob/branch-1.5/docs/README.md

File: docs/README.md [hunk 2]
```diff
@@ -32,42 +35,50 @@ the source code and be captured by revision control (currently git). This way th
...
+$ jekyll serve --watch
+
+# Build the site with extra features used on the live page
+$ PRODUCTION=1 jekyll build
 ```
 
-## API Docs (Scaladoc, Sphinx, roxygen2)
+## API Docs (Scaladoc, Javadoc, Sphinx, roxygen2, MkDocs)
 
-You can build just the Spark scaladoc by running `build/sbt unidoc` from the SPARK_PROJECT_ROOT directory.
```
Comment (HyukjinKwon) [inline docs/README.md]: I think we don't use the term `SPARK_PROJECT_ROOT` anywhere except for here. I rather chose the close term `SPARK_HOME`.

**Review by HyukjinKwon — COMMENTED** [review]
@srowen, I just decided to touch up the documentation around here little bit more aggressively while I am here. Please let me know if there is anything we should rather not touch for now or revert back.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**